envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/SourceIpBasedAuthenticationTest.kt

```diff
@@ -0,0 +1,73 @@
+package pl.allegro.tech.servicemesh.envoycontrol
```

Comment:
As a side note, do we have any test verifying `roles`? I only see `clients` in `incoming` in the test `proxy_settings`. Am I missing something?

Reply:
Look for function:
```
    fun `should generate RBAC rules for incoming permissions with roles`() {
```

Reply:
I see a unit test for the factory but I there are `IncomingPermissionsTest` and `SourceIpBasedAuthenticationTest` and none of them care about roles. That's why I'm asking :)

Reply:
We can add integration tests for it to cover more cases but I think that it's impossible to cover all the cases with integration tests. If you think it's necessary here I'll add it.